From 5236ce817db62e4d51d02b89d6c14b1e3edd7e6b Mon Sep 17 00:00:00 2001
From: Sandor Yu <Sandor.yu@nxp.com>
Date: Mon, 10 Sep 2018 15:40:57 +0800
Subject: [PATCH 4569/5242] MLK-19496-3: HDP: max pixel clock support rate
 297MHz for no edid case

commit  4d28db02e8a6daee4ebb569a2cc31395d2ec236e from
https://source.codeaurora.org/external/imx/linux-imx.git

Generally DDC function is not work with no edid.
Add max pixel clock support rate 297MHz for no edid case.

Signed-off-by: Sandor Yu <Sandor.yu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/hdp/imx-hdp.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/imx/hdp/imx-hdp.c b/drivers/gpu/drm/imx/hdp/imx-hdp.c
index fa065ce..da4079f 100644
--- a/drivers/gpu/drm/imx/hdp/imx-hdp.c
+++ b/drivers/gpu/drm/imx/hdp/imx-hdp.c
@@ -819,11 +819,14 @@ static int imx_hdp_connector_get_modes(struct drm_connector *connector)
 
 	cmdline_mode = &connector->cmdline_mode;
 
-	/* cmdline mode is the max support video mode when edid disabled */
+	/* cmdline mode is the max support video mode when edid disabled
+	 * Add max support pixel rate to 297MHz in case no DDC function with no EDID */
 	if (hdp->no_edid) {
 		if (cmdline_mode->xres != 0 &&
 			cmdline_mode->xres < mode->hdisplay)
 			return MODE_BAD_HVALUE;
+		if (mode->clock > 297000)
+			return MODE_CLOCK_HIGH;
 	}
 
 	/* For iMX8QM A0 Max support video mode is 4kp30 */
-- 
1.7.9.5

